lib/galaxy/tools/parameters/basic.py

```diff
@@ -1250,6 +1251,7 @@ def get_options(self, trans, other_values):
         if self.usecolnames:  # read first row - assume is a header with metadata useful for making good choices
             dataset = other_values.get(self.data_ref, None)
             try:
+                StorageMedia.refresh_all_media_credentials(dataset.active_storage_media_associations, self.app.authnz_manager, self.sa_session)
```

Comment:
I mean ... if we do decide that we want to always call this "refresh" thing before we use the dataset methods, we can do a lot better in terms of abstractions I think. 

```
self.app.dataset_manager.ensure_dataset_ready(dataset)
```

would probably be a better abstraction. We wouldn't need to import StorageMedia, basic.py wouldn't need to know about dataset.active_storage_media_associations or authnz at all. ``app`` has its own ``self.sa_session`` so we wouldn't need to pass it. One argument versus three would be a more maintainable interface.

Reply:
As per conversation, the cloud-based backends support are dropped for clarity. They will be made as a separate PR after this is merged.